src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Serialization.cs

```diff
@@ -57,60 +57,87 @@ internal partial class SymbolTreeInfo : IObjectWritable
             Func<ObjectReader, T> tryReadObject,
             CancellationToken cancellationToken) where T : class, IObjectWritable, IChecksummedObject
         {
-            if (checksum == null) 
+            Func<Task<T>> wrappedTask = async () =>
             {
-                return loadOnly ? null : await createAsync().ConfigureAwait(false);
-            }
+                using (Logger.LogBlock(FunctionId.SymbolTreeInfo_Create, cancellationToken))
+                {
+                    return await createAsync().ConfigureAwait(false);
+                }
+            };
```

Comment:
cleaner as just a local function?

Reply:
Yes, it seems like the new worker function is unnecessary. Just update both previous `createAsync()` calls to instead call a local function.